README.md

```diff
@@ -93,3 +93,94 @@ extension BannerView: BannerRendering {
 }
 ```
 
+## 🛰 Interactor
+
+> **Interactor**
+> (*noun*)
+> 
+> Component answering the question: What should I do in reaction to user action. 
+
+### Definitions
+
+### TL;DR
+
+Interactor is a final class that ViewController owns, which performs actions in reaction to user input - gestures, text input, device movement, geographical location change etc.
+
+#### Interactor is: 
+- active component (has a lifecycle), 
+- final class (1),
+- can use other components for data retrieval etc.
+- should render methods on weakly held views.
+
+*(1) There is no good reason for inheritance of custom classes, ever.*
+
+#### ViewControllers:
+
+- can have many Interactors
+- should OWN interactors and see them via protocol,
+
+### Example
+
+**Step 1 - Create Interactor protocols:**
+
+```swift
+protocol PodBayDoorsInteracting: class {
+	func use(_ banner: BannerRendering)
+	func didTapMainButton()
+}
+```
+
+**Step 2 - Add Interactor implementation:**
+
+```swift
+final class PodBayDoorsInteractor {
+	// Why FU? We do want to crash if use() is not called before banner is used.
```

Comment:
This makes `PodBayDoorsInteractor's` interface more semantic then programmatic. I wish you could avoid it, potentially by making `banner` dependency a part of initializer contract.

💡 Side note - this will also crash if `banner` reference is improperly managed beyond `PodBayDoorsInteractor` - if for some reason the `banner` is deallocated and `Interactor` not.

Reply:
I agree but creation and lifecycle of those rendering components (well, Views/Scenes/Documents) are out of our control but by crashing we can spot the error, everything should be called on main thread so no multithreading is an issue, you just called render before you handled him the view.

There is more - Interactor cannot be created before it's owner and have it's component handled to it via init.

Reply:
Fair enough, I've also seen the initialisation problem and couldn't find anything to advise.